### PR TITLE
strategy: factorzoo: refactor to logistic regression

### PR DESCRIPTION
**Pull request by austin362667**
- `divergence` *factorzoo.AVD // amplitude volume divergence
- `reversion`  *factorzoo.PMR // price mean reversion
- `momentum`   *factorzoo.MOM // price momentum from WorldQuant's paper, alpha 101
- `drift`      *indicator.Drift // from GBM
- `volume`     *factorzoo.VMOM  // quarterly volume momentum
- `bars`       *factorzoo.LSBAR // long short bar accumulation

### CODE REVIEW

**Comment by austin362667** [comment]
截圖 2022-07-21 下午4 23 42

**Comment by austin362667** [comment]
```
symbol: BTCUSDT
winningRatio: "1.14285714"
numOfLossTrade: 105
numOfProfitTrade: 120
grossProfit: "257184.89999993"
grossLoss: "-172114.42999993"

mostProfitableTrade: "7332.00000000"
mostLossTrade: "-6617.33000000"
profitFactor: "1.49426692"
totalNetProfit: "85070.47000000"
```

File: config/factorzoo.yaml [hunk 1]
```diff
@@ -9,22 +9,48 @@ exchangeStrategies:
...
+          side: sell
+
+## (0) roiStopLoss is the stop loss percentage of the position ROI (currently the price change)
+#      - roiStopLoss:
+#          percentage: 1.0%
+#
+## (1) roiTakeProfit is used to force taking profit by percentage of the position ROI (currently the price change)
+## force to take the profit ROI exceeded the percentage.
+#      - roiTakeProfit:
+#          percentage: 35%
```
Comment (c9s) [inline config/factorzoo.yaml:42]: remove these unused comment? you can create a `localconfig` directory for local testing config,

File: pkg/strategy/factorzoo/factors/amplitude_volume_divergence.go [hunk 2]
```diff
@@ -7,29 +7,56 @@ import (
...
-	Values  types.Float64Slice
+
+	// Values
+	Values    types.Float64Slice
+	LastValue float64
+
 	EndTime time.Time
 
-	UpdateCallbacks []func(value float64)
+	UpdateCallbacks []func(val float64)
```
Comment (c9s) [inline pkg/strategy/factorzoo/factors/amplitude_volume_divergence.go:29]: make this field private

File: pkg/strategy/factorzoo/factors/amplitude_volume_divergence.go [hunk 3]
```diff
@@ -7,29 +7,56 @@ import (
...
+func (inc *AVD) Length() int {
+	if inc.Values == nil {
+		return 0
+	}
+	return inc.Values.Length()
+}
+
+//var _ types.SeriesExtend = &AVD{}
+
+func (inc *AVD) Update(klines []types.KLine) {
```
Comment (c9s) [inline pkg/strategy/factorzoo/factors/amplitude_volume_divergence.go:55]: this function should be called `CalculateAndUpdate`, check the latest EWMA indicator's interface:

BindK()
PushK()
LoadK()

File: pkg/strategy/factorzoo/factors/long_short_bar.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	types.SeriesBase
+	types.IntervalWindow
+
+	// Values
+	Values    types.Float64Slice
+	LastValue float64
+
+	EndTime time.Time
+
+	UpdateCallbacks []func(val float64)
```
Comment (c9s) [inline pkg/strategy/factorzoo/factors/long_short_bar.go:24]: ditto

File: pkg/strategy/factorzoo/factors/momentum.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+	"time"
+
+	"github.com/c9s/bbgo/pkg/indicator"
+	"github.com/c9s/bbgo/pkg/types"
+
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate callbackgen -type MOM
+type MOM struct {
```
Comment (c9s) [inline pkg/strategy/factorzoo/factors/momentum.go:14]: do you have a reference link for this MOM?

File: pkg/strategy/factorzoo/factors/return_rate.go [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+	"time"
+
+	"github.com/c9s/bbgo/pkg/indicator"
+	"github.com/c9s/bbgo/pkg/types"
+
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate callbackgen -type RR
+type RR struct {
```
Comment (c9s) [inline pkg/strategy/factorzoo/factors/return_rate.go:14]: And a reference link for this RR?

File: pkg/strategy/factorzoo/factors/volume_momentum.go [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+	"time"
+
+	"github.com/c9s/bbgo/pkg/indicator"
+	"github.com/c9s/bbgo/pkg/types"
+
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate callbackgen -type VMOM
+type VMOM struct {
```
Comment (c9s) [inline pkg/strategy/factorzoo/factors/volume_momentum.go:14]: and reference link for this VMOM?

**Pull request closed without merging**